src/main/java/sirpotato/Storage.java

```diff
@@ -0,0 +1,106 @@
+package sirpotato;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+import java.util.Scanner; 
+import java.util.ArrayList; 
+
+
+/**
+ * Storage class handles the reading and writing of tasks to the file
+ */
+public class Storage {
+
+    private String filePath;
+    private DateTimeFormatter saveFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    
+    public Storage(String filePath) {
+        this.filePath = filePath;
+
+        try {
+            //File f = new File("../../../data/list.txt"); 
+            File f = new File(filePath); 
+            if (f.createNewFile()) {
+                System.out.println("Creating your data file");
+            } else {
+                System.out.println("Initializing your data file");
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Converts date in text file to LocalDate object 
+     * @return LocalDate object in the dd-MM-yyyy format
+     */
+    private LocalDate parseData(String dateToParse) throws DateTimeParseException {
```

Comment:
Since this method focuses on parsing date, maybe the method can be named `parseDate` so it's more specific?